contracts/gateway/Gateway.sol

```diff
@@ -0,0 +1,202 @@
+     */
+    function deployIdentityWithSalt(
+        address identityOwner,
+        string memory salt,
+        uint256 signatureExpiry,
+        bytes calldata signature
+    ) external returns (address) {
+        if (identityOwner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (signatureExpiry != 0 && signatureExpiry < block.timestamp) {
+            revert ExpiredSignature();
+        }
+
+        address signer = ECDSA.recover(
+            keccak256(
+                abi.encode(
+                    "Authorize ONCHAINID deployment",
+                    identityOwner,
+                    salt,
+                    signatureExpiry
+                )
+            ).toEthSignedMessageHash(),
+            signature
+        );
+
+        if (!approvedSigners[signer]) {
+            revert UnapprovedSigner();
+        }
+
+        if (revokedSignatures[signature]) {
+            revert RevokedSignature();
+        }
+
+        return idFactory.createIdentity(identityOwner, salt);
+    }
+
+    /**
+     *  @dev Deploy an ONCHAINID using a factory using the identityOwner address as salt.
+     *  @param identityOwner the address to set as a management key.
+     */
+    function deployIdentityForWallet(address identityOwner) external returns (address) {
+        if (identityOwner == address(0)) {
+            revert ZeroAddress();
+        }
+        if (identityOwner != msg.sender) {
+            revert UnapprovedSigner();
+        }
+
+        return idFactory.createIdentity(identityOwner, Strings.toHexString(identityOwner));
+    }
+
+    /**
+     *  @dev Revoke a signature, if the signature is used to deploy an ONCHAINID, the deployment would be rejected.
+     *  @param signature the signature to revoke.
+     */
+    function revokeSignature(bytes calldata signature) external onlyOwner {
+        if (revokedSignatures[signature]) {
+            revert SignatureAlreadyRevoked();
+        }
+
+        revokedSignatures[signature] = true;
+
+        emit SignatureRevoked(signature);
+    }
+
+    /**
+     *  @dev Remove a signature from the revoke list.
+     *  @param signature the signature to approve.
+     */
+    function approveSignature(bytes calldata signature) external onlyOwner {
+        if (!revokedSignatures[signature]) {
+            revert SignatureNotRevoked();
+        }
+
+        delete revokedSignatures[signature];
+
+        emit SignatureApproved(signature);
+    }
+
+    /**
+     *  @dev Transfer the ownership of the factory to a new owner.
+     *  @param newOwner the new owner of the factory.
+     */
+    function transferFactoryOwnership(address newOwner) external onlyOwner {
+        idFactory.transferOwnership(newOwner);
+    }
+
+    /**
+     *  @dev Call a function on the factory. Only the owner of the Gateway can call this method.
+     *  @param data the data to call on the factory.
+     */
+    function callFactory(bytes memory data) external onlyOwner {
```

Comment:
if we have this owner scoped function for all calls on Factory, then the `transferFactoryOwnership` could be removed, no? As ownership transfer can be called from `callFactory`